service/src/main/java/datawave/microservice/query/QueryManagementService.java

```diff
@@ -0,0 +1,2602 @@
+     * <p>
+     * Created queries will start running immediately. <br>
+     * Auditing is performed before the query is started. <br>
+     * Subsequent query results can be retrieved using {@link #executeNext}. <br>
+     * Updates can be made to any parameter which doesn't affect the scope of the query using {@link #update}. <br>
+     * Stop a running query gracefully using {@link #close} or forcefully using {@link #cancel}. <br>
+     * Stop, and restart a running query using {@link #reset}. <br>
+     * Create a copy of a running query using {@link #duplicate}. <br>
+     * Aside from a limited set of admin actions, only the query owner can act on a running query.
+     *
+     * @param queryLogicName
+     *            the requested query logic, not null
+     * @param parameters
+     *            the query parameters, not null
+     * @param pool
+     *            the pool to target, may be null
+     * @param currentUser
+     *            the user who called this method, not null
+     * @return a base query response containing the first page of results
+     * @throws BadRequestQueryException
+     *             if parameter validation fails
+     * @throws BadRequestQueryException
+     *             if query logic parameter validation fails
+     * @throws UnauthorizedQueryException
+     *             if the user doesn't have access to the requested query logic
+     * @throws BadRequestQueryException
+     *             if security marking validation fails
+     * @throws BadRequestQueryException
+     *             if auditing fails
+     * @throws QueryException
+     *             if query storage fails
+     * @throws TimeoutQueryException
+     *             if the next call times out
+     * @throws NoResultsQueryException
+     *             if no query results are found
+     * @throws QueryException
+     *             if this next task is rejected by the executor
+     * @throws QueryException
+     *             if there is an unknown error
+     */
+    public BaseQueryResponse createAndNext(String queryLogicName, MultiValueMap<String,String> parameters, String pool, DatawaveUserDetails currentUser)
+                    throws QueryException {
+        String user = ProxiedEntityUtils.getShortName(currentUser.getPrimaryUser().getName());
+        if (log.isDebugEnabled()) {
+            log.info("Request: {}/createAndNext from {} with params: {}", queryLogicName, user, parameters);
+        } else {
+            log.info("Request: {}/createAndNext from {}", queryLogicName, user);
+        }
+        
+        String queryId = null;
+        try {
+            queryId = create(queryLogicName, parameters, pool, currentUser).getResult();
+            return executeNext(queryId, currentUser);
+        } catch (Exception e) {
+            // TODO: QueryException results in VoidResponse. This differs from the original RestAPI where an exception was returned in a
```

Comment:
A note that this TODO is still here

Reply:
I think this particular TODO is something that we will need to address once the query microservices are vetted, before we transition from using wildfly to using all microservices.  We aren't quite at that point yet, so my hope is that this won't hold up the approval process.  Does that sound fair?

Reply:
cleared out all of the TODOs that I added.